src/Makefile

```diff
@@ -325,10 +325,29 @@ ifndef V
         @printf '    %b %b\n' $(LINKCOLOR)INSTALL$(ENDCOLOR) $(BINCOLOR)$(1)$(ENDCOLOR) 1>&2
         @$(INSTALL) $(1) $(2)
     endef
+
+    define INSTALL_REDIS_SYMLINK
+        @printf '    %b %b %b %b %b %b %b\n' \
+        $(LINKCOLOR)INSTALL SYMLINK$(ENDCOLOR) \
+        $(BINCOLOR)$(subst $(ENGINE_NAME),redis,$(1))$(ENDCOLOR) -\> $(BINCOLOR)$(1)$(ENDCOLOR) 1>&2
+        @ln -sf $(1) $(2)/$(subst $(ENGINE_NAME),redis,$(1))
+    endef
 else
     define MAKE_INSTALL
         $(INSTALL) $(1) $(2)
     endef
+
+    define INSTALL_REDIS_SYMLINK
+        ln -sf $(1) $(2)/$(subst $(ENGINE_NAME),redis,$(1))
+    endef
+endif
+
+ifeq ($(USE_REDIS_SYMLINKS),no)
+	MAYBE_INSTALL_REDIS_SYMLINK=
+	MAYBE_UNINSTALL_REDIS_SYMLINK=
+else
+	MAYBE_INSTALL_REDIS_SYMLINK=$(INSTALL_REDIS_SYMLINK)
+	MAYBE_UNINSTALL_REDIS_SYMLINK=@rm -f $(1)/$(subst $(ENGINE_NAME),redis,$(2))
 endif
```

Comment:
Very nice! Setting both variables in the same ifeq is better than my suggestion.

Just maybe we should check for ifneq "yes" instead of ifeq "no"? If we check for "no", then anything else will be treated as yes, which might be unexpected..? I don't know if it matters though.

Reply:
If use `ifeq yes` it has to be after `USE_REDIS_SYMLINKS?=yes`.
I think related functions like `define INSTALL_REDIS_SYMLINK` and `MAYBE_INSTALL_REDIS_SYMLINK` should be defined close together for clarity.

Maybe I should move the `USE_REDIS_SYMLINKS?=yes` code.

Reply:
I'm not sure but I think you can use a variable before it's defined. The semantics of variables in make is interesting.

Looks good to move it anyway. It seems more logical.